src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs

```diff
@@ -278,11 +282,37 @@ public static SignedPackageArchiveMetadata ReadSignedArchiveMetadata(BinaryReade
             metadata.EndOfCentralDirectory = lastCentralDirectoryRecord.Position + lastCentralDirectoryRecord.HeaderSize;
             metadata.CentralDirectoryHeaders = centralDirectoryRecords;
 
-            UpdateSignedPackageArchiveMetadata(reader, metadata);
+            var endOfLocalFileHeadersPosition = centralDirectoryRecords.Min(record => record.Position);
+
+            metadata.EndOfFileHeaders = endOfLocalFileHeadersPosition;
+
+            UpdateLocalFileHeadersTotalSize(centralDirectoryRecords, endOfLocalFileHeadersPosition);
 
             return metadata;
         }
 
+        private static void UpdateLocalFileHeadersTotalSize(
```

Comment:
By making this change, are the functions `SeekReaderBackwardToMatchByteSignature` and `SeekReaderForwardToMatchByteSignature` still being used? If not, it should be safe to delete them.